Bugzilla/Attachment.pm

```diff
@@ -542,13 +542,22 @@ sub get_attachments_by_bug {
   my $dbh  = Bugzilla->dbh;
 
   # By default, private attachments are not accessible, unless the user
-  # is in the insider group or submitted the attachment.
+  # is in the insider group, submitted the attachment, or it's a bounty
+  # attachment and they reported the bug.
   my $and_restriction = '';
   my @values          = ($bug->id);
 
   unless ($user->is_insider) {
-    $and_restriction = 'AND (isprivate = 0 OR submitter_id = ?)';
+    $and_restriction = 'AND (isprivate = 0 OR submitter_id = ?';
     push(@values, $user->id);
+    if ($user->id == $bug->{reporter_id}) {
+      # Keep these conditions in sync with _attachment_is_bounty_attachment
+      # in extensions/BMO/Extension.pm
+      $and_restriction .= ' OR (filename = ? AND mimetype = ?)';
+      push(@values, 'bugbounty.data');
+      push(@values, 'text/plain');
+    }
```

Comment:
Since the values are string literals and not variables, we can just put the values in the SQL directly.

```
$and_restriction .= " OR (filename = 'bugbounty.data' AND mimetype = 'text/plain')";
```